error raised when `select * from client_errors_summary_by_user limit ? `

## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

### 2. What did you expect to see? (Required)
No error 

### 3. What did you see instead (Required)
error logs raised on tidb-0
``` go
2023-12-15 09:56:43	
[2023/12/15 01:56:43.185 +00:00] [INFO] [conn.go:1184] ["command dispatched failed"] [conn=2130679811808554449] [connInfo="id:2130679811808554449, addr:10.250.173.206:15957 status:10, collation:utf8mb4_0900_ai_ci, user:root"] [command=Query] [status="inTxn:0, autocommit:1"] [sql="select * from `client_errors_summary_by_user` limit ?"] [txn_mode=PESSIMISTIC] [timestamp=0] [err="runtime error: invalid memory address or nil pointer dereference\ngithub.com/pingcap/tidb/executor.(*recordSet).Next.func1\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/executor/adapter.go:147\nruntime.gopanic\n\t/usr/local/go/src/runtime/panic.go:884\nruntime.panicmem\n\t/usr/local/go/src/runtime/panic.go:260\nruntime.sigpanic\n\t/usr/local/go/src/runtime/signal_unix.go:841\ngithub.com/pingcap/tidb/executor.(*memtableRetriever).setDataForClientErrorsSummary\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/executor/infoschema_reader.go:2372\ngithub.com/pingcap/tidb/executor.(*memtableRetriever).retrieve\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/executor/infoschema_reader.go:165\ngithub.com/pingcap/tidb/executor.(*MemTableReaderExec).Next\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/executor/memtable_reader.go:117\ngithub.com/pingcap/tidb/executor.Next\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/executor/executor.go:326\ngithub.com/pingcap/tidb/executor.(*LimitExec).Next\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/executor/executor.go:1463\ngithub.com/pingcap/tidb/executor.Next\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/executor/executor.go:326\ngithub.com/pingcap/tidb/executor.(*ExecStmt).next\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/executor/adapter.go:1202\ngithub.com/pingcap/tidb/executor.(*recordSet).Next\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/executor/adapter.go:151\ngithub.com/pingcap/tidb/server.(*tidbResultSet).Next\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/server/driver_tidb.go:469\ngithub.com/pingcap/tidb/server.(*clientConn).writeChunks\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/server/conn.go:2313\ngithub.com/pingcap/tidb/server.(*clientConn).writeResultSet\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/server/conn.go:2256\ngithub.com/pingcap/tidb/server.(*clientConn).handleStmt\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/server/conn.go:2124\ngithub.com/pingcap/tidb/server.(*clientConn).handleQuery\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/server/conn.go:1885\ngithub.com/pingcap/tidb/server.(*clientConn).dispatch\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/server/conn.go:1372\ngithub.com/pingcap/tidb/server.(*clientConn).Run\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/server/conn.go:1153\ngithub.com/pingcap/tidb/server.(*Server).onConn\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/server/server.go:677\nruntime.goexit\n\t/usr/local/go/src/runtime/asm_amd64.s:1598"]
2023-12-15 09:56:02	
[2023/12/15 01:56:02.322 +00:00] [INFO] [conn.go:1184] ["command dispatched failed"] [conn=2130679811808554449] [connInfo="id:2130679811808554449, addr:10.250.173.206:15957 status:10, collation:utf8mb4_0900_ai_ci, user:root"] [command=Query] [status="inTxn:0, autocommit:1"] [sql="select count ( ? ) from `client_errors_summary_by_user`"] [txn_mode=PESSIMISTIC] [timestamp=0] [err="runtime error: invalid memory address or nil pointer dereference\ngithub.com/pingcap/tidb/executor.recoveryHashAgg\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/executor/aggregate.go:457\ngithub.com/pingcap/tidb/executor.(*HashAggExec).fetchChildData.func1\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/executor/aggregate.go:803\nruntime.gopanic\n\t/usr/local/go/src/runtime/panic.go:884\nruntime.panicmem\n\t/usr/local/go/src/runtime/panic.go:260\nruntime.sigpanic\n\t/usr/local/go/src/runtime/signal_unix.go:841\ngithub.com/pingcap/tidb/executor.(*memtableRetriever).setDataForClientErrorsSummary\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/executor/infoschema_reader.go:2372\ngithub.com/pingcap/tidb/executor.(*memtableRetriever).retrieve\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/executor/infoschema_reader.go:165\ngithub.com/pingcap/tidb/executor.(*MemTableReaderExec).Next\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/executor/memtable_reader.go:117\ngithub.com/pingcap/tidb/executor.Next\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/executor/executor.go:326\ngithub.com/pingcap/tidb/executor.(*HashAggExec).fetchChildData\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/executor/aggregate.go:821\nruntime.goexit\n\t/usr/local/go/src/runtime/asm_amd64.s:1598"]
2023-12-15 09:56:02	
[2023/12/15 01:56:02.322 +00:00] [ERROR] [aggregate.go:458] ["parallel hash aggregation panicked"] [error="runtime error: invalid memory address or nil pointer dereference"] [stack="github.com/pingcap/tidb/executor.recoveryHashAgg\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/executor/aggregate.go:458\ngithub.com/pingcap/tidb/executor.(*HashAggExec).fetchChildData.func1\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/executor/aggregate.go:803\nruntime.gopanic\n\t/usr/local/go/src/runtime/panic.go:884\nruntime.panicmem\n\t/usr/local/go/src/runtime/panic.go:260\nruntime.sigpanic\n\t/usr/local/go/src/runtime/signal_unix.go:841\ngithub.com/pingcap/tidb/executor.(*memtableRetriever).setDataForClientErrorsSummary\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/executor/infoschema_reader.go:2372\ngithub.com/pingcap/tidb/executor.(*memtableRetriever).retrieve\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/executor/infoschema_reader.go:165\ngithub.com/pingcap/tidb/executor.(*MemTableReaderExec).Next\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/executor/memtable_reader.go:117\ngithub.com/pingcap/tidb/executor.Next\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/executor/executor.go:326\ngithub.com/pingcap/tidb/executor.(*HashAggExec).fetchChildData\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/executor/aggregate.go:821"]
2023-12-15 09:56:02	
[2023/12/15 01:56:02.281 +00:00] [INFO] [conn.go:1184] ["command dispatched failed"] [conn=2130679811808554449] [connInfo="id:2130679811808554449, addr:10.250.173.206:15957 status:10, collation:utf8mb4_0900_ai_ci, user:root"] [command=Query] [status="inTxn:0, autocommit:1"] [sql="select `user` , `error_number` , `error_message` , `error_count` , `warning_count` , `first_seen` , `last_seen` from `client_errors_summary_by_user` limit ..."] [txn_mode=PESSIMISTIC] [timestamp=0] [err="runtime error: invalid memory address or nil pointer dereference\ngithub.com/pingcap/tidb/executor.(*recordSet).Next.func1\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/executor/adapter.go:147\nruntime.gopanic\n\t/usr/local/go/src/runtime/panic.go:884\nruntime.panicmem\n\t/usr/local/go/src/runtime/panic.go:260\nruntime.sigpanic\n\t/usr/local/go/src/runtime/signal_unix.go:841\ngithub.com/pingcap/tidb/executor.(*memtableRetriever).setDataForClientErrorsSummary\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/executor/infoschema_reader.go:2372\ngithub.com/pingcap/tidb/executor.(*memtableRetriever).retrieve\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/executor/infoschema_reader.go:165\ngithub.com/pingcap/tidb/executor.(*MemTableReaderExec).Next\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/executor/memtable_reader.go:117\ngithub.com/pingcap/tidb/executor.Next\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/executor/executor.go:326\ngithub.com/pingcap/tidb/executor.(*LimitExec).Next\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/executor/executor.go:1463\ngithub.com/pingcap/tidb/executor.Next\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/executor/executor.go:326\ngithub.com/pingcap/tidb/executor.(*ExecStmt).next\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/executor/adapter.go:1202\ngithub.com/pingcap/tidb/executor.(*recordSet).Next\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/executor/adapter.go:151\ngithub.com/pingcap/tidb/server.(*tidbResultSet).Next\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/server/driver_tidb.go:469\ngithub.com/pingcap/tidb/server.(*clientConn).writeChunks\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/server/conn.go:2313\ngithub.com/pingcap/tidb/server.(*clientConn).writeResultSet\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/server/conn.go:2256\ngithub.com/pingcap/tidb/server.(*clientConn).handleStmt\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/server/conn.go:2124\ngithub.com/pingcap/tidb/server.(*clientConn).handleQuery\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/server/conn.go:1885\ngithub.com/pingcap/tidb/server.(*clientConn).dispatch\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/server/conn.go:1372\ngithub.com/pingcap/tidb/server.(*clientConn).Run\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/server/conn.go:1153\ngithub.com/pingcap/tidb/server.(*Server).onConn\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/server/server.go:677\nruntime.goexit\n\t/usr/local/go/src/runtime/asm_amd64.s:1598"]
2023-12-15 09:56:00	
[2023/12/15 01:56:00.472 +00:00] [INFO] [conn.go:1184] ["command dispatched failed"] [conn=2130679811808554449] [connInfo="id:2130679811808554449, addr:10.250.173.206:15957 status:10, collation:utf8mb4_0900_ai_ci, user:root"] [command=Query] [status="inTxn:0, autocommit:1"] [sql="select count ( ? ) from `client_errors_summary_by_user`"] [txn_mode=PESSIMISTIC] [timestamp=0] [err="runtime error: invalid memory address or nil pointer dereference\ngithub.com/pingcap/tidb/executor.recoveryHashAgg\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/executor/aggregate.go:457\ngithub.com/pingcap/tidb/executor.(*HashAggExec).fetchChildData.func1\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/executor/aggregate.go:803\nruntime.gopanic\n\t/usr/local/go/src/runtime/panic.go:884\nruntime.panicmem\n\t/usr/local/go/src/runtime/panic.go:260\nruntime.sigpanic\n\t/usr/local/go/src/runtime/signal_unix.go:841\ngithub.com/pingcap/tidb/executor.(*memtableRetriever).setDataForClientErrorsSummary\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/executor/infoschema_reader.go:2372\ngithub.com/pingcap/tidb/executor.(*memtableRetriever).retrieve\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/executor/infoschema_reader.go:165\ngithub.com/pingcap/tidb/executor.(*MemTableReaderExec).Next\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/executor/memtable_reader.go:117\ngithub.com/pingcap/tidb/executor.Next\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/executor/executor.go:326\ngithub.com/pingcap/tidb/executor.(*HashAggExec).fetchChildData\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/executor/aggregate.go:821\nruntime.goexit\n\t/usr/local/go/src/runtime/asm_amd64.s:1598"]
2023-12-15 09:56:00	
[2023/12/15 01:56:00.472 +00:00] [ERROR] [aggregate.go:458] ["parallel hash aggregation panicked"] [error="runtime error: invalid memory address or nil pointer dereference"] [stack="github.com/pingcap/tidb/executor.recoveryHashAgg\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/executor/aggregate.go:458\ngithub.com/pingcap/tidb/executor.(*HashAggExec).fetchChildData.func1\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/executor/aggregate.go:803\nruntime.gopanic\n\t/usr/local/go/src/runtime/panic.go:884\nruntime.panicmem\n\t/usr/local/go/src/runtime/panic.go:260\nruntime.sigpanic\n\t/usr/local/go/src/runtime/signal_unix.go:841\ngithub.com/pingcap/tidb/executor.(*memtableRetriever).setDataForClientErrorsSummary\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/executor/infoschema_reader.go:2372\ngithub.com/pingcap/tidb/executor.(*memtableRetriever).retrieve\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/executor/infoschema_reader.go:165\ngithub.com/pingcap/tidb/executor.(*MemTableReaderExec).Next\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/executor/memtable_reader.go:117\ngithub.com/pingcap/tidb/executor.Next\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/executor/executor.go:326\ngithub.com/pingcap/tidb/executor.(*HashAggExec).fetchChildData\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/executor/aggregate.go:821"]
```

### 4. What is your TiDB version? (Required)

v7.1.1

--- a/pkg/domain/infosync/error.go
+++ b/pkg/domain/infosync/error.go
@@ -16,13 +16,10 @@ package infosync
 
 import (
 	"github.com/pingcap/tidb/pkg/errno"
-	"github.com/pingcap/tidb/pkg/parser/mysql"
 	"github.com/pingcap/tidb/pkg/util/dbterror"
 )
 
 var (
 	// ErrHTTPServiceError means we got a http response with a status code which is not '2xx'
-	ErrHTTPServiceError = dbterror.ClassDomain.NewStdErr(
-		errno.ErrHTTPServiceError, mysql.Message("HTTP request failed with status %s", nil),
-	)
+	ErrHTTPServiceError = dbterror.ClassDomain.NewStd(errno.ErrHTTPServiceError)
 )

--- a/pkg/errno/BUILD.bazel
+++ b/pkg/errno/BUILD.bazel
@@ -16,10 +16,12 @@ go_test(
     name = "errno_test",
     timeout = "short",
     srcs = [
+        "errname_test.go",
         "infoschema_test.go",
         "main_test.go",
     ],
     embed = [":errno"],
+    embedsrcs = ["errcode.go"],
     flaky = True,
     deps = [
         "//pkg/testkit/testsetup",

--- a/pkg/errno/errcode.go
+++ b/pkg/errno/errcode.go
@@ -1068,7 +1068,6 @@ const (
 	ErrLoadDataInvalidURI                  = 8158
 	ErrLoadDataCantAccess                  = 8159
 	ErrLoadDataCantRead                    = 8160
-	ErrLoadDataPhysicalImportTableNotEmpty = 8161
 	ErrLoadDataWrongFormatConfig           = 8162
 	ErrUnknownOption                       = 8163
 	ErrInvalidOptionVal                    = 8164

--- a/pkg/errno/errname.go
+++ b/pkg/errno/errname.go
@@ -1073,6 +1073,7 @@ var MySQLErrName = map[uint16]*mysql.ErrMessage{
 	ErrLoadDataPreCheckFailed:           mysql.Message("PreCheck failed: %s", nil),
 	ErrMemoryExceedForQuery:             mysql.Message("Your query has been cancelled due to exceeding the allowed memory limit for a single SQL query. Please try narrowing your query scope or increase the tidb_mem_quota_query limit and try again.[conn=%d]", nil),
 	ErrMemoryExceedForInstance:          mysql.Message("Your query has been cancelled due to exceeding the allowed memory limit for the tidb-server instance and this query is currently using the most memory. Please try narrowing your query scope or increase the tidb_server_memory_limit and try again.[conn=%d]", nil),
+	ErrHTTPServiceError:                 mysql.Message("HTTP request failed with status %s", nil),
 
 	ErrWarnOptimizerHintInvalidInteger:  mysql.Message("integer value is out of range in '%s'", nil),
 	ErrWarnOptimizerHintUnsupportedHint: mysql.Message("Optimizer hint %s is not supported by TiDB and is ignored", nil),

--- a/pkg/errno/errname_test.go
+++ b/pkg/errno/errname_test.go
@@ -1,0 +1,47 @@
+// Copyright 2023 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package errno
+
+import (
+	_ "embed"
+	"strconv"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+//go:embed errcode.go
+var errCodeSrc string
+
+func TestAllErrCodeHasMsg(t *testing.T) {
+	lines := strings.Split(errCodeSrc, "\n")
+	errCodes := make([]uint16, 0, len(lines))
+	for _, l := range lines {
+		l = strings.TrimSpace(l)
+		if !strings.HasPrefix(l, "Err") {
+			continue
+		}
+		codeStr := strings.TrimSpace(strings.Split(l, "=")[1])
+		code, err := strconv.Atoi(codeStr)
+		assert.NoErrorf(t, err, "parse code definition: %s", codeStr)
+		errCodes = append(errCodes, uint16(code))
+	}
+
+	for _, code := range errCodes {
+		_, ok := MySQLErrName[code]
+		assert.Truef(t, ok, "ErrCode: %d is unknown", code)
+	}
+}

